src/hotspot/share/cds/filemap.cpp

```diff
@@ -1904,15 +1907,26 @@ bool FileMapInfo::relocate_pointers_in_core_regions(intx addr_delta) {
   if (bitmap_base == nullptr) {
     return false; // OOM, or CRC check failure
   } else {
-    size_t ptrmap_size_in_bits = header()->ptrmap_size_in_bits();
-    log_debug(cds, reloc)("mapped relocation bitmap @ " INTPTR_FORMAT " (" SIZE_FORMAT " bits)",
-                          p2i(bitmap_base), ptrmap_size_in_bits);
+    size_t rw_ptrmap_size_in_bits = header()->rw_ptrmap_size_in_bits();
+    size_t ro_ptrmap_size_in_bits = header()->ro_ptrmap_size_in_bits();
 
-    BitMapView ptrmap((BitMap::bm_word_t*)bitmap_base, ptrmap_size_in_bits);
+    BitMapView rw_ptrmap((BitMap::bm_word_t*)bitmap_base, rw_ptrmap_size_in_bits);
+    char* ro_bitmap_base = bitmap_base + rw_ptrmap.size_in_bytes();
+    BitMapView ro_ptrmap((BitMap::bm_word_t*)ro_bitmap_base, ro_ptrmap_size_in_bits);
```

Comment:
There's no guarantee that `FileMap::write_bitmap` will actually write only `rw_ptrmap.size_in_bytes()`, as it could do some padding as well.

If we use the `init_oopmap()` call as suggested above, then the above two lines can be replaced by a call to `FileMapRegion::bitmap_view(false)`.

We should do the same for the RW region as well. This way, we can delegate the BitMapView creation to common code.